enclave_wrangler/dataset_download_new_api.py

```diff
@@ -12,67 +13,116 @@
 from enclave_wrangler.config import config
 # from enclave_wrangler.utils import log_debug_info
 
+
 HEADERS = {
     "authorization": f"Bearer {config['PALANTIR_ENCLAVE_AUTHENTICATION_BEARER_TOKEN']}",
-    #'content-type': 'application/json'
+    # 'content-type': 'application/json'
 }
 DEBUG = False
 # TARGET_CSV_DIR='data/datasets/'
 
 
-@typechecked
-def objTypes() -> [{}]:
-    """API documentation at
-    https://www.palantir.com/docs/foundry/api/ontology-resources/object-types/list-object-types/
-    """
-
-    ontologyRid = config['ONTOLOGY_RID']
-    api_path = f'/api/v1/ontologies/{ontologyRid}/objectTypes'
-    url = f'https://{config["HOSTNAME"]}{api_path}'
-
-    response = requests.get(url, headers=HEADERS,)
-    response_json = response.json()
-    # print(response_json)
-    # types = pd.DataFrame(data=response_json)
-    return response_json['data']
-
-
-def run() -> None:
+class EnclaveClient:
+    """REST client for N3C data enclave"""
+
+    def __init__(self):
+        self.headers = HEADERS
+        self.debug = DEBUG
+        self.base_url = f'https://{config["HOSTNAME"]}'
+        self.ontology_rid = config['ONTOLOGY_RID']
+
+    # TODO: @Siggie: I fetched data but couldn't find concept sets. Here's what I found that looked interesting:
+    #  ConceptRelationship, ConceptSetBundleItem, ConceptSetTag, ConceptSetVersionChangeAcknowledgement,
+    #  ConceptSuccessorRelationship, ConceptUsageCounts, CsetVersionInfo, CodeSystemConceptSetVersionExpressionItem,
+    #  OMOPConcept, OMOPConceptAncestorRelationship, OMOPConceptChangeConnectors, OMOPConceptSet,
+    #  OMOPConceptSetContainer, OMOPConceptSetReview, OmopConceptChange, OmopConceptDomain, OmopConceptSetVersionItem,
+    @typechecked
+    def obj_types(self) -> List[Dict]:
+        """Gets object types.
+        API docs: https://www.palantir.com/docs/foundry/api/ontology-resources/object-types/list-object-types/"""
+        url = f'{self.base_url}/api/v1/ontologies/{self.ontology_rid}/objectTypes'
+        response = requests.get(url, headers=self.headers)
+        response_json = response.json()['data']  # always returns everything in nested 'data' key
+        # types = pd.DataFrame(data=response_json)
+        # types = sorted([x['apiName'] for x in response_json])
+        return response_json['data']
+
+    # TODO: Implement this func to get all concept sets
+    #  - handle pagination. I think get 1k at a time
+    # TODO: after: for each set, get all concepts. Siggie said an API call for that is already in TermHub code i think
+    # TODO: Need to find the right object_type, then write a wrapper func around this to get concept sets
+    #  - To Try: CodeSystemConceptSetVersionExpressionItem, OMOPConcept, OMOPConceptSet, OMOPConceptSetContainer,
+    #    OmopConceptSetVersionItem
+    def objects(self, object_type: str = 'OMOPConceptSet'):  # todo: temp setting to single type until find
```

Comment:
@Sigfried FYI, was surprised that the page tokens are so similarly formatted. They must be a concatenation of previously generated hashes. The pattern I found was that each one starts with:
`v1.eyJ0IjoiT2JqZWN0cy4xIiwicHMiOjEwMDAsInRva2VuIjoidjEuZXlKeVpYRjFaWE4wUTJobFkydHpkVzBpT2lJd00yUmtOVFV4WkdFNFl6QXpOakUxTXpKalpUTmtNelZtWXpSaVptUmlZbUpqTlRnNU1XVXpNVEV5TkRFd09USTROV016TUdVeFlqRmlZek13TURWaUlpd2lZbUZqYTJWdVpGQmhaMlZVYjJ0bGJpSTZJbll4TGpNeU56TXhaVFU0TFRRME5tWXRORFU0WkMwNE5UUXdMVFZqT0RRNFl6QTJOalUyTkM0e`

Then is followed by two characters.

Then finally ends with:
`EQXdJaXdpWW1GamEyVnVaQ0k2SWxCSVQwNVBSMUpCVUVnaWZRPT0ifQ==`